app/experimenter/static/js/components/TimelinePopForm.js

```diff
@@ -94,6 +96,53 @@ class TimelinePopForm extends React.PureComponent {
     }
   }
 
+  displayEnrollmentDuration() {
+    if (this.props.experimentType != "rollout") {
+      return (
+        <DesignInput
+          label="Proposed Enrollment Duration (days)"
+          name="proposed_enrollment"
+          id="id_proposed_enrollment"
+          onChange={value => {
+            this.handleDataChange("proposed_enrollment", value);
+          }}
+          value={this.state.data.get("proposed_enrollment")}
+          error={this.state.errors.get("proposed_enrollment", "")}
+          helpContent={PROPOSED_ENROLLMENT_HELP}
+          labelColumnWidth={2}
+          optional={true}
+        />
+      );
+    }
+  }
+
+  displayRolloutPlaybook() {
+    if (this.props.experimentType === "rollout") {
+      return (
+        <DesignInput
+          as="select"
+          label="Rollout Playbook"
+          name="rollout_playbook"
+          id="id_rollout_playbook"
+          onChange={value => {
+            this.handleDataChange("rollout_playbook", value);
+          }}
+          value={this.state.data.get("rollout_playbook")}
+          error={this.state.errors.get("rollout_playbook", "")}
+          labelColumnWidth={2}
+          helpContent={ROLLOUT_PLAYBOOK_HELP}
+          helpIsExternalLink={true}
+        >
+          <option>Rollout Playbook</option>
+          <option>Low Risk Schedule</option>
+          <option>High Risk Schedule</option>
+          <option>Marketing Launch Schedule</option>
+          <option>Custom Schedule</option>
```

Comment:
Oh shouldn't these be 

1) in the js constants file
2) those are the display strings but don't the options also need the string values (the lower case ones) here? https://github.com/mozilla/experimenter/blob/master/app/experimenter/experiments/constants.py#L25-L35

Reply:
you'd like to see if more how I did the firefox versions?

Reply:
Because I just put the value into every option and that works, however I can refactor it to look more like firefox versions, where there's a function that format's each option withwhat's in the JS constants file.

Reply:
Oooohhh I see what's happening here.

The fields in the database are charfields, so they'll accept anything, so anything you put in there is valid and will just get saved and displayed.

Usually in Django world you define two things, a db value and a display value, and then define the  pair as `CHOICES`, example:

https://github.com/mozilla/experimenter/blob/master/app/experimenter/experiments/constants.py#L10-L20

Then later when you're dealing with the field in code somewhere, you're comparing  against the db value not the display value.

We didn't notice because a bunch of these things just only stay as text and we never refer to teh values in code, we just take wahtever it is and display it on  the detail page, and the db values and display values are the same, like for platform:

https://github.com/mozilla/experimenter/blob/master/app/experimenter/experiments/constants.py#L152-L157

So yes exactly it should be like the way versions are done, and actually the real fix here is to change the serializer to use a ChoiceField instead of a CharField so it actually checks that the values are from the list we expect, but I think we can do that as a followup ticket so we don't blow this one up.

Reply:
Followup: https://github.com/mozilla/experimenter/issues/2407

Reply:
ok I've made the change to where the playbook options are in constants and then they are displayed with the function `displayOptions`, which is a generalized version of a function that used to just be formatting the versions options... but is now formatting versions and playbook options.